onnx/__init__.py

```diff
@@ -140,24 +154,28 @@ def _deserialize(s: bytes, proto: _Proto) -> _Proto:
     if not (hasattr(proto, "ParseFromString") and callable(proto.ParseFromString)):
         raise TypeError(f"No ParseFromString method is detected. Type is {type(proto)}")
 
-    decoded = typing.cast(Optional[int], proto.ParseFromString(s))
-    if decoded is not None and decoded != len(s):
-        raise google.protobuf.message.DecodeError(
-            f"Protobuf decoding consumed too few bytes: {decoded} out of {len(s)}"
-        )
+    if format_ == "textproto":
+        google.protobuf.text_format.Parse(s.decode("utf-8"), proto)
+    else:
+        assert format_ == "protobuf"
```

Comment:
I would raise an exception with a better error message. It is likely to be seen by users when they call save_model.

Reply:
Ok!

Reply:
Done